lib/modules/platform/bitbucket-server/utils.ts

```diff
@@ -148,3 +150,19 @@ export function getExtraCloneOpts(opts: HostRule): GitOptions {
   }
   return {};
 }
+
+// https://docs.atlassian.com/bitbucket-server/rest/5.1.0/bitbucket-rest.html#idm45588158982432
+export async function resolveEmailToUsername(
+  http: BitbucketServerHttp,
+  email: string,
+): Promise<string | null> {
+  // GET /rest/api/1.0/admin/users?filter={filter}
+  const users = (
+    await http.getJsonUnchecked<RestDetailedUser[]>(
+      `./rest/api/1.0/admin/users?filter=${email}`,
+      { paginate: true },
```

Comment:
```suggestion
    await http.getJsonUnchecked(
      `./rest/api/1.0/admin/users?filter=${email}`,
      { paginate: true },
      z.array(RestDetailedUser),
```

Reply:
you can also precreate the userarray in schema.ts to not need it import zod here